core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableJoinRule.java

```diff
@@ -61,12 +61,29 @@
     if (!info.isEqui() && join.getJoinType() != JoinRelType.INNER) {
       // EnumerableHashJoin only supports equi-join. We can put a filter on top
       // if it is an inner join.
-      return EnumerableNestedLoopJoin.create(
-          left,
-          right,
-          join.getCondition(),
-          join.getVariablesSet(),
-          join.getJoinType());
+      RelNode newRel;
+      //if it has equiKeys ,we can create an EnumerableHashJoin with
```

Comment:
Minor detail. Please review style in the comment line: whitespace after '//' and after comma (not before). Same applies for comment in line 74.